app/src/main/java/com/example/mytrainingpal/components/GifComponent.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.mytrainingpal.components
+
+import android.os.Build
+import androidx.compose.foundation.Image
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import coil.ImageLoader
+import coil.compose.rememberAsyncImagePainter
+import coil.decode.GifDecoder
+import coil.decode.ImageDecoderDecoder
+import coil.request.ImageRequest
+
+// Reference: Dr BigKitkat answer on github + Coil library docs
+// https://stackoverflow.com/questions/60229555/adding-gif-into-jetpack-compose
+// about AsyncImagePainter --> https://coil-kt.github.io/coil/compose/
+
+
+@Composable
+fun GifImage(
+    size: Int,
+    // TODO: gifs saved under R.drawable.<gifName> (added through the GUI Resource Manager)
+    gifPath: Any?
+) {
+    val context = LocalContext.current
+    val imageLoader = ImageLoader.Builder(context)
```

Comment:
Here is how to get the resource id of the gif given by path to be able to render it. Put all exercise gifs into the main/res/drawable folder and rename them so they do not contain capital letters -> `Exercise_1.gif` becomes `exercise_1.gif`. If you now pass `gifPath = "exercise_1.gif` to the `GifImage` composable, the gif should be shown.
```suggestion
    // Getting the resource id of the gif using its path.
    // Code from https://stackoverflow.com/questions/70062705/how-can-i-get-drawable-resource-by-string was used.
    val gifResourceId = remember(gifPath) {
      context.resources.getIdentifier(
          gifPath,
          "drawable",
          context.packageName
      )
  }
    val imageLoader = ImageLoader.Builder(context)
```